internal/status/conditions.go

```diff
@@ -65,53 +67,54 @@ func (routeUpdate *HTTPRouteUpdate) AddCondition(cond gatewayapi_v1alpha1.RouteC
 	return newDc
 }
 
-// HTTPRouteAccessor returns a HTTPRouteUpdate that allows a client to build up a list of
+// ConditionsAccessor returns a ConditionsUpdate that allows a client to build up a list of
 // metav1.Conditions as well as a function to commit the change back to the cache when everything
-// is done. The commit function pattern is used so that the HTTPRouteUpdate does not need
+// is done. The commit function pattern is used so that the ConditionsUpdate does not need
 // to know anything the cache internals.
-func (c *Cache) HTTPRouteAccessor(route *gatewayapi_v1alpha1.HTTPRoute) (*HTTPRouteUpdate, func()) {
-	pu := &HTTPRouteUpdate{
-		FullName:           k8s.NamespacedNameOf(route),
+func (c *Cache) ConditionsAccessor(nsName types.NamespacedName, generation int64, resource string, gateways []gatewayapi_v1alpha1.RouteGatewayStatus) (*ConditionsUpdate, func()) {
```

Comment:
might be cool to have a generic route type interface in the upstream api that could be used to get the status info and concrete route type etc., maybe a more generic k8s object meta type could be used to get the generation etc., possible refactors for the future